src/makeCalculator.js

```diff
@@ -4,7 +4,50 @@
  * @return {object}
  */
 function makeCalculator() {
-  // write code here
+  const calculator = {
+    result: 0,
+
+    add(x) {
+      this.result += x;
+
+      return this;
+    },
+
+    subtract(x) {
+      this.result -= x;
+
+      return this;
+    },
+
+    multiply(x) {
+      this.result *= x;
+
+      return this;
+    },
+
+    divide(x) {
+      if (x === 0) {
+        throw new Error('Cannot divide by zero');
+      }
+      this.result /= x;
+
+      return this;
+    },
+
+    reset() {
+      this.result = 0;
+
+      return this;
+    },
+
+    operate(operation, number) {
+      operation.call(this, number);
+
+      return this;
```

Comment:
The `operate` method uses `operation.call(this, number)`, which assumes that the `operation` argument is a function. It might be safer to check if `operation` is indeed a function before calling it to prevent runtime errors. Consider adding a check like `if (typeof operation === 'function')` before calling it.